src/qibolab/instruments/zhinst/sweep.py

```diff
@@ -0,0 +1,139 @@
+"""Pre-execution processing of sweeps."""
+
+from collections.abc import Iterable
+from copy import copy
+
+import laboneq.simple as lo
+
+from qibolab.pulses import Pulse, PulseType
+from qibolab.qubits import Qubit
+from qibolab.sweeper import Parameter, Sweeper
+
+from .util import NANO_TO_SECONDS, measure_channel_name
+
+
+def classify_sweepers(
+    sweepers: Iterable[Sweeper],
+) -> tuple[list[Sweeper], list[Sweeper]]:
+    """Divide sweepers into two lists: 1. sweeps that can be done in the laboneq near-time sweep loop, 2. sweeps that
+    can be done in real-time (i.e. on hardware)"""
+    nt_sweepers, rt_sweepers = [], []
+    for sweeper in sweepers:
+        if (
+            sweeper.parameter is Parameter.amplitude
+            and sweeper.pulses[0].type is PulseType.READOUT
+        ):
+            nt_sweepers.append(sweeper)
+        elif sweeper.parameter is Parameter.bias:
+            nt_sweepers.append(sweeper)
+        else:
+            rt_sweepers.append(sweeper)
+    return nt_sweepers, rt_sweepers
```

Comment:
I'd split the function in two: the condition (it could return just a boolean), and the collection.

Reply:
hmm, I am not sure I understood this one. What do you mean by "the condition"?

Reply:
The only boolean condition distinguishing near-time from real-time, i.e.
```py
            sweeper.parameter is Parameter.bias or
            (sweeper.parameter is Parameter.amplitude
            and sweeper.pulses[0].type is PulseType.READOUT)
```

Reply:
ah, I see. I wouldn't actually try to clean this up too much. Because 
1. there is no forecast that the condition is going to get more complicated soon, 
3. this whole concepts of classifying sweepers is going to change drastically with introduction of logical channels. The condition is not going to exist as it is now. Currently is kind of a holistic condition trying to cover every case, in reality each channel should do this separately on its own.